context/python/parse.py

```diff
@@ -7,62 +7,73 @@
 from sets import Sets
 
 
+def arg_parser():
+    parser = argparse.ArgumentParser(
+        description='A variety of set intersection visualizations')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--json', type=str,
+        help='If given file does not exist, fall back to environment variable')
+    group.add_argument(
+        '--lists', nargs='+', type=argparse.FileType('r'),
+        help='Single column lists')
+
+    parser.add_argument(
+        '--output', type=str,
+        help='Destination directory')
+
+    return parser
+
+
 def get_input_json(possible_input_file):
     '''
     Checks three possible sources for JSON input,
     and returns JSON string if found.
     Returns None if all three fail.
     '''
-    if possible_input_file and os.path.isfile(possible_input_file):
+    if os.path.isfile(possible_input_file):
         return open(possible_input_file, 'r').read(None)
 
     json = os.environ.get('INPUT_JSON')
     if json:
+        print('INPUT_JSON: ' + json[:40])
         return json
 
     url = os.environ.get('INPUT_JSON_URL')
     if url:
-        print('url: ' + url)
+        print('INPUT_JSON_URL: ' + url)
         return requests.get(url).text
 
-    return None
+    raise Exception('No input.json from any source')
 
-def load(input_json_path):
-    input_json = get_input_json(input_json_path)
-    if input_json:
-        data = json.loads(input_json)
-        strings = [requests.get(url).text for url in data['file_relationships']]
-        sets = []
 
-def arg_parser():
-    parser = argparse.ArgumentParser(
-        description='A variety of set intersection visualizations')
-
-    parser.add_argument(
-        '--json', type=argparse.FileType('r'),
-        help='input.json')
-    parser.add_argument(
-        '--lists', nargs='+', type=argparse.FileType('r'),
-        help='Single column lists')
-    parser.add_argument(
-        '--output', type=str,
-        help='Destination directory'
-    )
+def read_json(input_json_path):
+    input_json = get_input_json(input_json_path)
+    data = json.loads(input_json)
+    return {
+        os.path.basename(url): {
+            l.strip() for l in requests.get(url).text.split('\n')
+        } for url in data['file_relationships']
```

Comment:
I think using the `'file_relationships'` key here is fine if we just care about getting the data files, but will fall short if we want the associated Refinery metadata as well. Other key is called: `node_info`

Ex:
```
"node_info": {
    "7b8c6585-cdca-4274-9df0-6906b15e27b2": {
      "file_url": "http://192.168.50.50:8000/media/file_store/e9/9c/s7_EV_E2_45min.subsample_RUqHcz6.fastq",
      "node_solr_info": {
        "REFINERY_SUBANALYSIS_339_186_s": "-1",
        "REFINERY_NAME_339_186_s": "s7_EV_E2_45min.subsample.fastq",
        "data_set_uuid": "75f86f59-40c3-49f8-b383-57da6f28eccb",
        "django_ct": "data_set_manager.node",
        "REFINERY_FILETYPE_339_186_s": "FASTQ file",
        "sample_id_Characteristics_339_186_s": "s7_EV_E2_45min.subsample",
        "is_annotation": false,
        "REFINERY_DOWNLOAD_URL_s": "/media/file_store/e9/9c/s7_EV_E2_45min.subsample_RUqHcz6.fastq",
        "id": "data_set_manager.node.3094",
        "organism_Characteristics_generic_s": "Human",
        "django_id": "3094",
        "organism_Characteristics_339_186_s": "Human",
        "uuid": "7b8c6585-cdca-4274-9df0-6906b15e27b2",
        "sample_id_Characteristics_generic_s": "s7_EV_E2_45min.subsample",
        "assay_uuid": "f1971781-ab98-47e9-b01a-2574e2f39ffd",
        "REFINERY_TYPE_339_186_s": "Raw Data File",
        "REFINERY_WORKFLOW_OUTPUT_339_186_s": "N/A",
        "REFINERY_DATAFILE_s": "e9/9c/s7_EV_E2_45min.subsample_RUqHcz6.fastq",
        "type": "Raw Data File",
        "is_metadata_revision_Characteristics_generic_s": "True",
        "is_metadata_revision_Characteristics_339_186_s": "True",
        "REFINERY_ANALYSIS_UUID_339_186_s": "N/A",
        "filetype_Characteristics_generic_s": "FASTQ file",
        "study_uuid": "59ade4dd-ee33-41a9-ab6d-45e2f726ad0d",
        "name": "s7_EV_E2_45min.subsample.fastq",
        "file_uuid": "b46f0889-c79b-4dcf-9124-be418c750293"
      }
    }
  }
```

Reply:
I don't have any plans to use the metadata... but would you prefer that all the tools look for their data in the same place? That would be an argument to change it...

Reply:
No strong argument from me either way